usaspending_api/submissions/management/commands/rm_submission.py

```diff
@@ -67,33 +29,19 @@ def signal_handler(signal, frame):
         signal.signal(signal.SIGINT, signal_handler)
         signal.signal(signal.SIGTERM, signal_handler)
 
-        # This will throw an exception and exit the command if the id doesn't exist
+        broker_submission_id = options["submission_id"][0]
+
         try:
-            submission = SubmissionAttributes.objects.get(broker_submission_id=options["submission_id"][0])
+            submission = SubmissionAttributes.objects.get(broker_submission_id=broker_submission_id)
         except ObjectDoesNotExist:
-            raise "Submission with broker id " + str(options["submission_id"][0]) + " does not exist"
-
-        self.logger.info('Getting childless submissions')
-        potentially_childless = self.instances_potentially_left_childless(submission)
-        print('done')
-
-        deleted = defaultdict(int)
-        deleted_total = 0
-        deletion_tally = submission.delete()
-        deleted_total += deletion_tally[0]
-        deleted.update(deletion_tally[1])
+            raise "Broker submission id {} does not exist".format(broker_submission_id)
 
-        self.logger.info('Starting iteration over potentially childless submissions')
-        for instance in potentially_childless:
-            deletions = instance.delete_if_childless()
-            deleted_total += deletions[0]
-            for (key, value) in deletions[1].items():
-                deleted[key] += value
+        deleted_stats = submission.delete()
 
-        self.logger.info('Finished deletions')
+        self.logger.info('Finished deletions.')
 
-        statistics = "Statistics:\n  Total objects Removed: {}".format(deleted_total)
-        for (model, count) in deleted.items():
-            statistics = statistics + "\n  {}: {}".format(model, count)
+        statistics = "Statistics:\n  Total objects removed: {}".format(deleted_stats[0])
+        for (model, count) in deleted_stats[1].items():
+            statistics += "\n  {}: {}".format(model, count)
```

Comment:
If deleted_stats is a large dict the string += will be costly. It is better to append to a list and then use `'\n '.join(temp_list)`

Reply:
This is expected to be < 10 lines. This can stay.

Reply:
no problem